### PR TITLE
Improved tests for a logging exporter and jaeger exporter

### PR DESCRIPTION
**Pull request by agneum**
Some improvements of tests for an issue https://github.com/open-telemetry/opentelemetry-collector/issues/266

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@agneum thank you for contributing! Please make sure the build passes and the maintainers will review the PR.

**Comment by agneum** [comment]
@tigrannajaryan Thank you, fixed

File: exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go [hunk 1]
```diff
@@ -57,21 +49,39 @@ func TestNew(t *testing.T) {
...
 
 			// This is expected to fail.
 			err = got.ConsumeTraceData(context.Background(), consumerdata.TraceData{})
 			assert.Error(t, err)
 		})
 	}
 }
+
+func TestNewFails(t *testing.T) {
+	tests := []struct {
```
Comment (tigrannajaryan) [inline exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go]: Since there is only one test here the value of having an array of test definitions is unclear here.

File: exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go [hunk 2]
```diff
@@ -57,21 +49,39 @@ func TestNew(t *testing.T) {
...
+				config:      nil,
+				httpAddress: testHTTPAddress,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := New(tt.args.config, tt.args.httpAddress, tt.args.headers, tt.args.timeout)
+			assert.EqualError(t, err, "nil config")
```
Comment (tigrannajaryan) [inline exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go]: This assumes all test cases in `tests` (if we had more than one) will return the same error message. This may not be true. Generally unclear why we have `tests` array and loop through it if it is one element.

File: exporter/jaeger/jaegerthrifthttpexporter/factory_test.go [hunk 3]
```diff
@@ -66,66 +66,81 @@ func TestCreateInstanceViaFactory(t *testing.T) {
 
 func TestFactory_CreateTraceExporter(t *testing.T) {
 	tests := []struct {
```
Comment (tigrannajaryan) [inline exporter/jaeger/jaegerthrifthttpexporter/factory_test.go]: Same here, `tests` with single element.

File: exporter/loggingexporter/logging_exporter_test.go [hunk 4]
```diff
@@ -28,28 +28,24 @@ import (
 
 func TestLoggingTraceExporterNoErrors(t *testing.T) {
 	lte, err := NewTraceExporter(&configmodels.ExporterSettings{}, zap.NewNop())
-	if err != nil {
-		t.Fatalf("Wanted nil got %v", err)
-	}
+	assert.NotNil(t, lte)
```
Comment (tigrannajaryan) [inline exporter/loggingexporter/logging_exporter_test.go]: This needs to use `require` instead of `assert` otherwise the test will crash if nil is returned. Same in other similar places.

**Comment by tigrannajaryan** [comment]
@agneum thank you for working on this. Please make sure `assert` or `require` is chosen according to what is needed in the particular piece of code.

**Comment by pjanotti** [comment]
kicking another build to see if we get coverage numbers for this PR

File: exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go (hunk 1 above)
Comment (agneum) [reply exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go]: I assumed that other cases could be added here in the future. But you're right, for now there is only one. So I fixed this test.

**Comment by agneum** [comment]
@tigrannajaryan Thank you for a review. I've fixed all mistakes. Could you please check it one more time?

File: exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go (hunk 1 above)
Comment (tigrannajaryan) [reply exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go]: Let's use the principle of YAGNI here. :-)
We generally aim to have minimum amount of code that serves the purpose.

File: exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go [hunk 5]
```diff
@@ -20,33 +20,26 @@ import (
...
-	}
+type args struct {
+	config      configmodels.Exporter
+	httpAddress string
+	headers     map[string]string
+	timeout     time.Duration
+}
+
+func TestNew(t *testing.T) {
 	tests := []struct {
```
Comment (tigrannajaryan) [inline exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go]: This one also is a single test case, let's remove the `tests`. It can be added later if there are multiple similar tests.

File: exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go [hunk 6]
```diff
@@ -57,21 +50,27 @@ func TestNew(t *testing.T) {
...
 			got, err := New(tt.args.config, tt.args.httpAddress, tt.args.headers, tt.args.timeout)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("New() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-
-			if got == nil {
-				return
-			}
+			assert.NoError(t, err, "nil config")
```
Comment (tigrannajaryan) [inline exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go]: The third parameter of NoError looks incorrect to me. If I remember correctly it is the message to print when assertion fails. This assertion is not about config, why does it print "nil config"?

Comment (agneum) [reply exporter/jaeger/jaegerthrifthttpexporter/exporter_test.go]: Indeed, it is a wrong usage. Fixed
